### PR TITLE
Replace quarkus.camel.native.resources.* with quarkus.native.resource…

### LINKED ISSUE #988: Replace quarkus.camel.native.resources.* with quarkus.native.resources.*
Native resources and reflection inclusion and exclusion is a very generic functionality that all Quarkus users could benefit from.

### PR DESCRIPTION
**Pull request by ppalaga**
…s.* fix #988

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Review by jamesnetherton — APPROVED** [review]
I guess we should do a scan of the example projects and clean this up over there too.

**Comment by ppalaga** [comment]
> I guess we should do a scan of the example projects and clean this up over there too.

Indeed, I wanted to, ... and forgotten. Thanks for the reminder!

**Comment by ppalaga** [comment]
Fun fact: `quarkus.native.resources.includes` honors trailing white space in the value, while `quarkus.camel.native.resources.include-patterns` used to strip it. That's why the stringtemplate test failed.

File: docs/modules/ROOT/pages/migration-guide/2.0.0.adoc [hunk 1]
```diff
@@ -19,3 +19,8 @@ The empty `camel-quarkus-main` artifact was removed completely in Camel Quarkus
 
 As long as your application depends on any other Camel Quarkus extension, it is enough to remove the `camel-quarkus-main` dependency.
 This is because all Camel Quarkus extensions transitively depend on `camel-quarkus-core` where the the original `camel-quarkus-main` functionality is hosted now.
+
+== `quarkus.camel.native.resources.*` replaced by `quarkus.native.resources.includes`
+
+The `quarkus.camel.native.resources.include-patterns` and `quarkus.camel.native.resources.exclude-patterns` configuration properties were removed in Camel Quarkus 2.0.0.
+Please use `quarkus.native.resources.includes` and `quarkus.native.resources.includes` instead respectively.
```
Comment (aldettinger) [inline docs/modules/ROOT/pages/migration-guide/2.0.0.adoc:26]: Maybe a typo ?
includes and quarkus.native.resources.**excludes** instead

File: extensions-core/core/runtime/src/main/java/org/apache/camel/quarkus/core/CamelConfig.java [hunk 2]
```diff
@@ -212,25 +212,15 @@
...
-         * resources not selected by quarkus itself are ignored. Then, inclusion
-         * of additional resources could be triggered with
-         * <code>includePatterns</code>. When the inclusion patterns is too
-         * large, eviction of previously selected resources could be triggered
-         * with <code>excludePatterns</code>.
+         * Replaced by {@code quarkus.native.resources.includes} in Camel Quarkus 2.0.0.
+         * Using this property throws an exception at build time.
          */
         @ConfigItem
         public Optional<List<String>> includePatterns;
```
Comment (aldettinger) [inline extensions-core/core/runtime/src/main/java/org/apache/camel/quarkus/core/CamelConfig.java:226]: My understanding is that we are going to break in camel-quarkus 2.0.0, so maybe we could remove those config items ?

File: extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/CamelNativeImageProcessor.java [hunk 3]
```diff
@@ -217,31 +217,13 @@ void embedSelectResourcesInNativeExecutable(CamelConfig config, ApplicationArchi
             BuildProducer<NativeImageResourceBuildItem> resources) {
 
         final ResourcesConfig resourcesConfig = config.native_.resources;
-        if (!resourcesConfig.includePatterns.isPresent()) {
-            LOGGER.debug("Not scanning resources for native inclusion as include-patterns is not set");
-            return;
+        if (resourcesConfig.includePatterns.isPresent()) {
+            throw new IllegalStateException(
+                    "quarkus.camel.native.resources.include-patterns configuration property was removed in Camel Quarkus 2.0.0. Use quarkus.native.resources.includes instead.");
```
Comment (aldettinger) [inline extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/CamelNativeImageProcessor.java:222]: If we break in camel-quarkus 2.0, then maybe it's a good candidate for the migration guide ?

File: integration-tests/geocoder/src/main/java/org/apache/camel/quarkus/component/geocoder/it/MockApiService.java [hunk 4]
```diff
@@ -21,8 +21,10 @@
 import javax.enterprise.context.ApplicationScoped;
 
 import com.google.maps.GeoApiContext;
+import io.quarkus.runtime.annotations.RegisterForReflection;
 
 @ApplicationScoped
+@RegisterForReflection(targets = GeoApiContext.Builder.class)
```
Comment (aldettinger) [inline integration-tests/geocoder/src/main/java/org/apache/camel/quarkus/component/geocoder/it/MockApiService.java:27]: It's actually really cleaner written that way. So, we can remove quarkus.camel.native.reflection at all ?

**Review by aldettinger — CHANGES_REQUESTED** [review]
Really good to see this aligned on quarkus features :+1:  I would just have few questions above.

File: docs/modules/ROOT/pages/migration-guide/2.0.0.adoc (hunk 1 above)
Comment (ppalaga) [reply docs/modules/ROOT/pages/migration-guide/2.0.0.adoc:26]: +1, thanks!

File: extensions-core/core/runtime/src/main/java/org/apache/camel/quarkus/core/CamelConfig.java (hunk 2 above)
Comment (ppalaga) [reply extensions-core/core/runtime/src/main/java/org/apache/camel/quarkus/core/CamelConfig.java:226]: Yep, it is there in `docs/modules/ROOT/pages/migration-guide/2.0.0.adoc`

File: extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/CamelNativeImageProcessor.java (hunk 3 above)
Comment (ppalaga) [reply extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/CamelNativeImageProcessor.java:222]: Yes, done in docs/modules/ROOT/pages/migration-guide/2.0.0.adoc

File: extensions-core/core/runtime/src/main/java/org/apache/camel/quarkus/core/CamelConfig.java (hunk 2 above)
Comment (aldettinger) [reply extensions-core/core/runtime/src/main/java/org/apache/camel/quarkus/core/CamelConfig.java:226]: So, could we remove the includePatterns field then ?

Comment (ppalaga) [reply extensions-core/core/runtime/src/main/java/org/apache/camel/quarkus/core/CamelConfig.java:226]: Ah, sorry, I have not understood initially. If we remove it, users just get a warning about an unknown property. With the changes in the Processor https://github.com/apache/camel-quarkus/pull/2813/files they get a real build time failure.

Comment (aldettinger) [reply extensions-core/core/runtime/src/main/java/org/apache/camel/quarkus/core/CamelConfig.java:226]: ok, got it now :+1:

**Review by aldettinger — APPROVED** [review]

File: integration-tests/geocoder/src/main/java/org/apache/camel/quarkus/component/geocoder/it/MockApiService.java (hunk 4 above)
Comment (ppalaga) [reply integration-tests/geocoder/src/main/java/org/apache/camel/quarkus/component/geocoder/it/MockApiService.java:27]: Unlike `@RegisterForReflection`, `quarkus.camel.native.reflection` supports wildcards so they are not fully interchangeable.

**Comment by ppalaga** [comment]
> > I guess we should do a scan of the example projects and clean this up over there too.
> 
> Indeed, I wanted to, ... and forgotten. Thanks for the reminder!

I checked and there were no occurrences of `quarkus.camel.native.resources.`

**Pull request merged**